### PR TITLE
Dev -> Master for 2.1.0 release

### PR DESCRIPTION
**Pull request by JoseEspinosa**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,62 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## v2.1.0 - [date]
```
Comment (JoseEspinosa) [inline CHANGELOG.md:6]: Update date when releasing!

File: modules/local/annotate_boolean_peaks.nf [hunk 2]
```diff
@@ -2,10 +2,10 @@ process ANNOTATE_BOOLEAN_PEAKS {
     tag "$meta.id"
     label 'process_low'
 
-    conda (params.enable_conda ? "conda-forge::sed=4.7" : null)
+    conda "conda-forge::sed=4.7"
```
Comment (sateeshperi) [inline modules/local/annotate_boolean_peaks.nf:5]: should the conda be reading from a `environment.yml` file ?

File: modules/local/star_genomegenerate.nf
Comment (sateeshperi) [inline modules/local/star_genomegenerate.nf:1]: why have local versions of these nf-core modules ?

File: nextflow.config [hunk 3]
```diff
@@ -207,32 +263,36 @@ env {
...
+    author          = """Espinosa-Carrasco J, Patel H, Wang C, Ewels P"""
     homePage        = 'https://github.com/nf-core/chipseq'
-    description     = 'ChIP-seq peak-calling and differential analysis pipeline.'
+    description     = """ChIP-seq peak-calling and differential analysis pipeline."""
     mainScript      = 'main.nf'
-    nextflowVersion = '!>=21.10.3'
-    version         = '2.0.0'
+    nextflowVersion = '!>=23.04.0'
+    version         = '2.1.0'
+    doi             = ''
```
Comment (sateeshperi) [inline nextflow.config:295]: forgot DOI or n/a ?

File: CHANGELOG.md (hunk 1 above)
Comment (sateeshperi) [reply CHANGELOG.md:6]: I also missed in mine.....how do you do this? just direct commit to dev ?

File: workflows/chipseq.nf [hunk 4]
```diff
@@ -92,73 +35,101 @@ include { FILTER_BAM_BAMTOOLS } from '../subworkflows/local/filter_bam_bamtools'
...
+ch_bamtools_filter_pe_config = file(params.bamtools_filter_pe_config)
+
+// Header files for MultiQC
+ch_spp_nsc_header           = file("$projectDir/assets/multiqc/spp_nsc_header.txt", checkIfExists: true)
+ch_spp_rsc_header           = file("$projectDir/assets/multiqc/spp_rsc_header.txt", checkIfExists: true)
+ch_spp_correlation_header   = file("$projectDir/assets/multiqc/spp_correlation_header.txt", checkIfExists: true)
+ch_peak_count_header        = file("$projectDir/assets/multiqc/peak_count_header.txt", checkIfExists: true)
+ch_frip_score_header        = file("$projectDir/assets/multiqc/frip_score_header.txt", checkIfExists: true)
+ch_peak_annotation_header   = file("$projectDir/assets/multiqc/peak_annotation_header.txt", checkIfExists: true)
+ch_deseq2_pca_header        = Channel.value(file("$projectDir/assets/multiqc/deseq2_pca_header.txt", checkIfExists: true))
```
Comment (bjlang) [inline workflows/chipseq.nf:75]: I can't spot a difference in the usage of this header vs e.g. the ch_peak_annotation_header. What's the reason for defining some as paths and some as value channels?

**Review by bjlang — APPROVED** [review]

**Review by JoseEspinosa — COMMENTED** [review]
> I also missed in mine.....how do you do this? just direct commit to dev ? 

I will open a small PR to `dev` instead, ping me if you need a review

**Review by sateeshperi — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -52,49 +59,60 @@ You can find numerous talks on the [nf-core events page](https://nf-co.re/events
...
+sample,fastq_1,fastq_2,antibody,control
+WT_BCATENIN_IP_REP1,BLA203A1_S27_L006_R1_001.fastq.gz,,BCATENIN,WT_INPUT_REP1
+WT_BCATENIN_IP_REP2,BLA203A25_S16_L001_R1_001.fastq.gz,,BCATENIN,WT_INPUT_REP2
+WT_BCATENIN_IP_REP2,BLA203A25_S16_L002_R1_001.fastq.gz,,BCATENIN,WT_INPUT_REP2
+WT_BCATENIN_IP_REP2,BLA203A25_S16_L003_R1_001.fastq.gz,,BCATENIN,WT_INPUT_REP2
+WT_BCATENIN_IP_REP3,BLA203A49_S40_L001_R1_001.fastq.gz,,BCATENIN,WT_INPUT_REP3
+WT_INPUT_REP1,BLA203A6_S32_L006_R1_001.fastq.gz,,,
+WT_INPUT_REP2,BLA203A30_S21_L001_R1_001.fastq.gz,,,
+WT_INPUT_REP2,BLA203A30_S21_L002_R1_001.fastq.gz,,,
+WT_INPUT_REP3,BLA203A31_S21_L003_R1_001.fastq.gz,,,
```
[the remark this answers is not in the document]
Comment (JoseEspinosa) [reply README.md]: This was already updated af88e70e79af2b7f198de38b8d714138cc119bf0

File: conf/modules.config [hunk 6]
```diff
@@ -522,20 +546,17 @@ if (!params.skip_plot_fingerprint) {
...
-    withName: 'MACS2_CALLPEAK' {
+    withName: 'MACS3_CALLPEAK' {
         ext.args   = [
             '--keep-dup all',
             params.narrow_peak          ? '' : "--broad --broad-cutoff ${params.broad_cutoff}",
             params.save_macs_pileup     ? '--bdg --SPMR' : '',
             params.macs_fdr             ? "--qvalue ${params.macs_fdr}" : '',
             params.macs_pvalue          ? "--pvalue ${params.macs_pvalue}" : '',
-            params.aligner == "chromap" ? "--format BAM" : ''
+            params.aligner == "chromap" ? "--format BAM" : '' //TODO check if not needed anymore with new chromap versions
```
[the remark this answers is not in the document]
Comment (JoseEspinosa) [reply conf/modules.config]: Was already removed here: 22f7e10d17fc08730ce25f94180317f09d68c8ea

File: workflows/chipseq.nf [hunk 7]
```diff
@@ -429,225 +425,78 @@ workflow CHIPSEQ {
     //
     // MODULE: Calculute genome size with khmer
     //
+    // TODO move to prepare genome
```
[the remark this answers is not in the document]
Comment (JoseEspinosa) [reply workflows/chipseq.nf:428]: Yes, opened an issue: https://github.com/nf-core/chipseq/issues/416

File: modules/local/star_genomegenerate.nf
Comment (JoseEspinosa) [reply modules/local/star_genomegenerate.nf:1]: The reason is [this](https://github.com/JoseEspinosa/nf-core-chipseq/blob/f47e8ed0680096be2e361d4b3b63bab697e497a1/modules/local/star_genomegenerate.nf#L5-L6): the indexes produced by the nf-core modules will be incompatible with iGenomes.

**Pull request merged**